### DISCUSSION TITLE
How to store Dio's 'RequestOptions' in Table

### DISCUSSION THREAD

**Discussion by sharoonamjid** [simolus3/drift#1365] [Q&A]
What I'm trying to achieve is to retry the request if internet connection is back. For that, first I send a request like this:

```
await _dio.post(requestUrl, data: jsonEncode(data));
```

if device is not connected to the internet, obviously this request would be failed. Once the request is failed I catch that error and the error also provides as useful `RequestOptions`. I need to store that `RequestOptions` from that error in the database table so that in later time when app is opened again and internet is connected I could then re-request it from that `RequestOptions` because it has all the request data like url, headers, options etc.
I want to be able to perform the following:

```
List<RequestOptions> failedRequests = await getRequestOptionsFromDb();

failedRequests.forEach((requestOptions) {
    // using `fetch` from Dio library because it can make a request from RequestionOptions
    await _dio.fetch(requestOptions);
});

```

Can you please guide me how can I store the `RequestOptions` in the database table? Thank you.

**Comment by simolus3** [comment]
You can use [type converters](https://moor.simonbinder.eu/docs/advanced-features/type_converters/) to store custom objects in the database.

As we already discussed, `CancelToken`s can't reasonable be persisted. You can serialize most other properties to JSON and store that in a text column by using a type converter.
